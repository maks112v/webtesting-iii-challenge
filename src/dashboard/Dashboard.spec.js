// Test away
import React from "react";
import { render, fireEvent, cleanup } from "react-testing-library";
import renderer from 'react-test-renderer'

import Dashboard from "./Dashboard";

describe("<Dashboard />", () => {
  afterEach(() => cleanup());
  it("renders Dashboard", () => {
    render(<Dashboard />);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("defaults to open and unlocked", () => {
    const { getByText } = render(<Dashboard />);

    const unlocked = getByText(/unlocked/i); 
    const open = getByText(/open/i); 
    expect(unlocked).toBeTruthy();
    expect(open).toBeTruthy();
  });
});
