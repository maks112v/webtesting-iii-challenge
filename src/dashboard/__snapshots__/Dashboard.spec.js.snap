// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lock-button"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="toggle-button"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
